### PR TITLE
Add test for xmlhttprequest

### PR DESCRIPTION
**Pull request by kangxu**

### CODE REVIEW

File: XMLHttpRequest/event-error.html [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+
+<div id="log"></div>
+
+<script>
+
+async_test(function (t) {
+  var client = new XMLHttpRequest();
+  client.onerror = t.step_func(function(e) {
```
Comment (hallvors) [inline XMLHttpRequest/event-error.html]: Note: A 404 Not Found response will not trigger XMLHttpRequest.onerror.

**Comment by hallvors** [comment]
(Please see Critic review for further details)

**Comment by hallvors** [comment]
@kangxu I hope the review comments in Critic are useful and that the suggestions are clear. I'm happy to merge this after some changes :)

**Comment by kangxu** [comment]
@hallvors  I have updated it to address your comments

**Comment by hallvors** [comment]
Thank you @kangxu !

Would you like to write a response-on-404.htm test, or should I? 

To write it, do something like 
1) Amend https://github.com/w3c/web-platform-tests/blob/master/XMLHttpRequest/resources/status.py to have a configurable content-type header (status.py?ctype=text/html&code=404&text=Not+Found&content=foo )
2) Write a test requesting URLs like status.py?ctype=text/json&code=404&text=Not+Found&content={"foo":"bar"} with .responseType set to 'json'
3) Assert that .response property is not a {"foo":"bar"} object (404 should prevent implementation from setting .response AFAIK)

If you want this task please add a new issue with the above details and comment that you'll handle it. Otherwise, I'll start working on this early next week or so. Thanks! ;)

**Pull request merged**